File: 백준/Silver/11047. 동전 0/동전 0.py
n, k = map(int, input().split())
cnt = 0
coin_arr = []

for i in range(n):
  coin = int(input())
  coin_arr.append(coin)

coin_arr.reverse()


for i in coin_arr:
  if k == 0:
    break
  if k >= i:
    cnt += k // i
    k -= i * (k // i)

print(cnt)

File: 백준/Silver/11047. 동전 0/동전 0.cc
#include <iostream>
using namespace std;

int main() {
  int n, k;
  cin >> n >> k;
  int cnt = 0;
  int coin_arr[n];

  for (int i = 0; i < n; i++)
    cin >> coin_arr[i];

  for (int i = n-1; i >= 0; i--) {
    if (k == 0)
      break;

    if (k >= coin_arr[i]) {
      cnt += k / coin_arr[i];
      k -= coin_arr[i] * (k / coin_arr[i]);
    }
  }

  cout << cnt << "\n";
}